### ISSUE TITLE
better style guide checking (properly parse bash vs regex-hit-and-miss)

### ISSUE THREAD

**Issue by mstreuhofer** [kvz/bash3boilerplate#84]
See discussion in PR #81

**Comment by zbeekman** [comment]
Some potentially helpful links:
 - https://github.com/vorpaljs/bash-parser

**Comment by mstreuhofer** [comment]
what about shfmt (https://github.com/mvdan/sh) ?
it seems to be opinionated as in there is no way to influence the acceptable style but at least it is readily available in homebrew

**Comment by kvz** [comment]
Looks promising at first glance, have you ran it against any code?

**Comment by mstreuhofer** [comment]
yes. strangely it wants to change all `if [[ $var = "x" ]]` to `if [[ $var == "x" ]]`. Other than that it just removes some empty lines.

**Comment by kenden** [comment]
Shellcheck could be nice too https://github.com/koalaman/shellcheck

**Comment by zbeekman** [comment]
@kenden we [already use shellcheck](https://github.com/kvz/bash3boilerplate/blob/0289532f187c6ebcac9dbc3b1be23fb54e1d683c/test/acceptance.sh#L230-L251). This is about style beyond linting/syntax checking 😉
